openvdb/openvdb/points/PointRasterizeFrustum.h

```diff
@@ -0,0 +1,1805 @@
+enum class RasterMode
+{
+    ACCUMULATE = 0,
+    MAXIMUM,
+    AVERAGE
+};
+
+
+/// @brief A camera class that provides an interface for camera motion blur when rasterizing
+class RasterCamera
+{
+public:
+    explicit RasterCamera(const math::Transform& transform);
+
+    bool isStatic() const;
+
+    void clear();
+    void appendTransform(const math::Transform&, float weight = 1.0f);
+
+    size_t size() const;
+
+    void simplify();
+
+    bool hasWeight(Index i) const;
+    float weight(Index i) const;
+
+    const math::Transform& transform(Index i) const;
+    const math::Transform& firstTransform() const;
+    const math::Transform& lastTransform() const;
+
+    void setShutter(float start, float end);
+    float shutterStart() const;
+    float shutterEnd() const;
+
+private:
+    std::deque<math::Transform> mTransforms;
+    std::deque<float> mWeights;
+    // default to 180 degree film shutter
+    float mShutterStart = -0.25f;
+    float mShutterEnd = 0.25f;
+}; // class RasterCamera
+
+
+/// @brief A group of shared settings to be used in the Volume Rasterizer
+/// @param scaleByVoxelVolume   scale particle contributions by the volume of the receiving voxel
+/// @param velocityAttribute    the name of the velocity attribute
+/// @param velocityMotionBlur   bake the point velocities into the volume
+/// @param clipToFrustum        if enabled and the transform is a frustum transform, eliminate
+///                             points whose position does not lie within the frustum
+/// @param clipBBox             an optional world-space bounding box to clip the points
+///                             during rasterization
+/// @param clipMask             an optional mask, each point samples the mask using a
+///                             nearest-neighbor sampling and is only rasterized if active
+/// @param invertMask           if mask is provided, only rasterize if sample is inactive
+/// @param framesPerSecond      the global value for frames / second for computing motion blur
+/// @param threaded             if enabled, use threading to accelerate rasterization
+/// @note rasterization can clip can using any combination of bounding box, mask and frustum
+struct FrustumRasterizerSettings
+{
+    FrustumRasterizerSettings() = delete;
+
+    explicit FrustumRasterizerSettings(const math::Transform& _transform)
+        : transform(new math::Transform(_transform))
+        , camera(_transform) { }
+
+    math::Transform::Ptr transform;
+    RasterCamera camera;
+    float threshold = 1e-6f;
+    bool scaleByVoxelVolume = false;
+    Name velocityAttribute = "v";
+    bool useRadius = false;
+    float radiusScale = 1.0f;
+    Name radiusAttribute = "pscale";
+    bool accurateFrustumRadius = false;
+    bool accurateSphereMotionBlur = false;
+    bool velocityMotionBlur = false;
+    float framesPerSecond = 24.0f;
+    int motionSamples = 2;
+    bool threaded = true;
+}; // struct FrustumRasterizerSettings
```

Comment:
lots of bools here - could reduce this size with:
```
    bool accurateFrustumRadius : 1, 
        accurateSphereMotionBlur : 1, 
        velocityMotionBlur : 1,
        threaded : 1,
        useRadius : 1,
        scaleByVoxelVolume : 1;
```

Doesn't look like this is copied though so is minor.